services/connector/commands/request_accounts_test.go

```diff
@@ -0,0 +1,129 @@
+package commands
+
+import (
+	"encoding/json"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/status-im/status-go/eth-node/types"
+	"github.com/status-im/status-go/params"
+	persistence "github.com/status-im/status-go/services/connector/database"
+	walletCommon "github.com/status-im/status-go/services/wallet/common"
+	"github.com/status-im/status-go/signal"
+)
+
+func TestFailToRequestAccountsWithMissingDAppFields(t *testing.T) {
+	db, close := SetupTestDB(t)
+	defer close()
+
+	cmd := &RequestAccountsCommand{AccountsCommand: AccountsCommand{Db: db}}
+
+	// Missing DApp fields
+	request, err := ConstructRPCRequest("eth_requestAccounts", []interface{}{}, nil)
+	assert.NoError(t, err)
+
+	result, err := cmd.Execute(request)
+	assert.Equal(t, ErrRequestMissingDAppData, err)
+	assert.Empty(t, result)
+}
+
+func TestRequestAccountsWithSignalTimout(t *testing.T) {
+	db, close := SetupTestDB(t)
+	defer close()
+
+	clientHandler := NewClientSideHandler()
+
+	cmd := &RequestAccountsCommand{
+		ClientHandler:   clientHandler,
+		AccountsCommand: AccountsCommand{Db: db},
+	}
+
+	request, err := prepareSendTransactionRequest(testDAppData, types.Address{0x01})
+	assert.NoError(t, err)
+
+	backupWalletResponseMaxInterval := WalletResponseMaxInterval
+	WalletResponseMaxInterval = 1 * time.Millisecond
+
+	_, err = cmd.Execute(request)
+	assert.Equal(t, ErrWalletResponseTimeout, err)
+	WalletResponseMaxInterval = backupWalletResponseMaxInterval
+}
+
+func TestRequestAccountsTwoTimes(t *testing.T) {
+	db, close := SetupTestDB(t)
+	defer close()
+
+	nm := NetworkManagerMock{}
+	nm.SetNetworks([]*params.Network{
+		{
+			ChainID: walletCommon.OptimismMainnet,
+			Layer:   2,
+		},
+		{
+			ChainID: walletCommon.EthereumMainnet,
+			Layer:   1,
+		},
+	})
+
+	clientHandler := NewClientSideHandler()
+
+	cmd := &RequestAccountsCommand{
+		ClientHandler:   clientHandler,
+		NetworkManager:  &nm,
+		AccountsCommand: AccountsCommand{Db: db},
+	}
+
+	request, err := ConstructRPCRequest("eth_requestAccounts", []interface{}{}, &testDAppData)
+	assert.NoError(t, err)
+
+	accountAddress := types.BytesToAddress(types.FromHex("0x6d0aa2a774b74bb1d36f97700315adf962c69fcg"))
```

Comment:
Is this is on purpose 😄 ?
Hexadecimal characters are 0-9 and a-f (or A-F)
If not, we should probably add some validation